test/discovery_failure_test.py

```diff
@@ -38,12 +49,10 @@ def delete_broken_import_file(self):
             except OSError as exc:
                 _log.error("Could not remove broken import file %s: %r" % (f, exc))
 
-    @class_setup
+    @T.class_setup_teardown
     def setup_import_file(self):
         self.create_broken_import_file()
-
-    @class_teardown
-    def teardown_import_file(self):
+        yield
```

Comment:
setup_teardown is equivalent to try finally without needing to write try finally